src/install.c

```diff
@@ -375,6 +375,10 @@ GHashTable* determine_target_install_group(void)
 
 
 		if (slot_list_contains(selected_root_slots, parent)) {
+			if (iterslot->readonly) {
+				g_debug("\tNot adding readonly slot");
```

Comment:
Should we have a more noticeable warning in this case?

Reply:
That would mean we do not allow readonly slots.

The continue here is simply to not add slots marked as 'readonly' to the mapping of classes to images. This means we will not generate a 'bootloader -> botloader.0' mapping if 'bootloader.0' is marked readonly.

Where this will fail is when someone comes with an bundle containing an image for the 'bootloader' class, then it will detect there is no mapping for this and exit with an error.